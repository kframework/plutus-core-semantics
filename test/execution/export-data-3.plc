(program

(module Nat (import) (export ((MyPair (Pair NotThis))) ())
  (data MyPair ()
        (Pair (integer) (integer))
        (NotThis))
)

(module Foo (import Nat) (export () ())
  (define result
    (case (con Nat.Pair 3 7)
          (Nat.Pair (x y) (builtin multiplyInteger x y))
          (Nat.NotThis ()  -1)))
  (define swapped
    (case (con Nat.Pair 1 2)
      (Nat.Pair (x y) (con Nat.Pair y x))
      (Nat.NotThis () (con Nat.Pair 1 1)))))

)
