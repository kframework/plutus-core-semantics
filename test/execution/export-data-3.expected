<T>
  <k>
    .
  </k>
  <lenv>
    .Map
  </lenv>
  <genv>
    Foo . result |-> 21
    Foo . swapped |-> ( con Nat . Pair 2  1  .ValList )
  </genv>
  <modules>
    ModuleCellSetItem ( <Module>
      <curr>
        0
      </curr>
      <name>
        Foo
      </name>
      <dataCons>
        .TyExpList
      </dataCons>
      <decl>
        .QualCList
      </decl>
      <importedModules>
        Nat  .ModNmList
      </importedModules>
      <exportTy>
        .TyExpList
      </exportTy>
      <exportTm>
        .QualCList
      </exportTm>
    </Module> ) ModuleCellSetItem ( <Module>
      <curr>
        0
      </curr>
      <name>
        Nat
      </name>
      <dataCons>
        ( MyPair ( Pair  NotThis  .ConList ) )  .TyExpList
      </dataCons>
      <decl>
        .QualCList
      </decl>
      <importedModules>
        .ModNmList
      </importedModules>
      <exportTy>
        ( MyPair ( Pair  NotThis  .ConList ) )  .TyExpList
      </exportTy>
      <exportTm>
        .QualCList
      </exportTm>
    </Module> )
  </modules>
</T>
